src/fbserver.c

```diff
@@ -283,47 +324,44 @@ struct cache_entry* find_shm(uint64_t paddr, uint64_t sig, size_t length) {
             p += c;
         }
 
-        char* cmd = "croutonfindnacl";
-        char* args[] = {cmd, arg1, arg2, NULL};
-        char buffer[256];
-        log(2, "Running %s %s %s", cmd, arg1, arg2);
-        c = popen2(cmd, args, NULL, 0, buffer, sizeof(buffer));
-        if (c <= 0) {
-            error("Error running helper.");
-            return NULL;
-        }
-        buffer[c < sizeof(buffer) ? c : (sizeof(buffer)-1)] = 0;
-        log(2, "Result: %s", buffer);
+        int sock;
+        struct sockaddr_un addr;
+
+        sock = socket(AF_UNIX, SOCK_STREAM, 0);
+        addr.sun_family = AF_UNIX;
+        strncpy(addr.sun_path, SOCKET_PATH, sizeof(addr.sun_path));
 
-        /* Parse PID:file output */
-        char* cut = strchr(buffer, ':');
-        if (!cut) {
-            error("No ':' in helper reply: %s.", cut);
+        if (connect(sock, (struct sockaddr *)&addr, offsetof(struct sockaddr_un,
```

Comment:
I just use sizeof(addr). Because according to the man page,

    The addrlen argument that describes the enclosing sockaddr_un structure should have
    a value of at least:
        offsetof(struct sockaddr_un, sun_path)+strlen(addr.sun_path)+1
    or, more simply, addrlen can be specified as sizeof(struct sockaddr_un).